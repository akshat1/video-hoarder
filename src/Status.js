/**
 * @enum {string} Status
 */
export default {
  Pending: 'pending',
  Running: 'running',
  Failed: 'failed',
  Succeeded: 'succeeded'
};
