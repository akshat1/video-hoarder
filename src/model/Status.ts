export enum Status {
  Failed = "failed",
  Paused = "paused",
  Pending = "pending",
  Running = "running",
  Succeeded = "succeeded",
}

export enum StatusFilterValue {
  All = "all",
  Failed = "failed",
  Paused = "paused",
  Pending = "pending",
  Running = "running",
  Succeeded = "succeeded",
}


export const hasStarted = (status: Status): boolean =>
  status === Status.Running || status === Status.Failed || status === Status.Succeeded || status === Status.Paused;

export const hasConcluded = (status: Status): boolean =>
  status === Status.Failed || status === Status.Succeeded;

export const isPending = (status: Status): boolean =>
  status === Status.Pending;
