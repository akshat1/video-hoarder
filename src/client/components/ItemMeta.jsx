import { hasStarted, Status } from "../../model/Status";
import ItemStatus from "./ItemStatus";
import { Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React, { Fragment } from "react";

const useStyle = makeStyles(theme => ({
  meta: {
    marginLeft: theme.spacing(1),
  },

  statusIcon: {
    verticalAlign: "middle",
  },
}));

const ItemMeta = (props) => {
  const classes = useStyle();
  const { item } = props;
  const {
    status,
    updatedAt,
  } = item;
  const dtUpdatedAt = new Date(updatedAt);

  return (
    <Fragment>
      <ItemStatus
        className={classes.statusIcon}
        status={status}
      />
        <Choose>
          <When condition={hasStarted(status)}>
            <Typography
              className={classes.meta}
              display="inline"
            >
              <Choose>
                <When condition={status === Status.Failed}>Failed</When>
                <When condition={status === Status.Paused}>Paused</When>
                <When condition={status === Status.Running}>Updated</When>
                <When condition={status === Status.Succeeded}>Completed</When>
              </Choose>
              {` at ${dtUpdatedAt.toLocaleTimeString()} on ${dtUpdatedAt.toLocaleDateString()}`}
            </Typography>
          </When>
          <Otherwise>
            <Typography
              className={classes.meta}
              display="inline"
            >
              {`Queued at ${dtUpdatedAt.toLocaleTimeString()} on ${dtUpdatedAt.toLocaleDateString()}`}
            </Typography>
          </Otherwise>
        </Choose>
    </Fragment>
  );
};

export default ItemMeta;
