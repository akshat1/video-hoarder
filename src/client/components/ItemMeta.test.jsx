import React from 'react';
import { shallow } from 'enzyme';
import { getFakeItem } from '../fixtures/item';
import Status from '../../Status';
import ItemMeta from './ItemMeta';

describe('components/ItemMeta', () => {
  Object.values(Status).forEach((status) => {
    test(`ItemMeta matches snapshot for ${status}`, () => {
      expect(shallow(<ItemMeta item={getFakeItem(status)} />)).toMatchSnapshot();
    });
  });
});
