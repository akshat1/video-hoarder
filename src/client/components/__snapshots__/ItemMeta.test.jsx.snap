// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ItemMeta ItemMeta matches snapshot for failed 1`] = `
<table
  className="wrapper"
>
  <tbody>
    <tr>
      <td>
        Added
      </td>
      <td>
        3/29/2020, 2:30:45 PM
      </td>
    </tr>
    <tr>
      <td>
        Failed
      </td>
      <td>
        3/29/2020, 2:32:30 PM
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`components/ItemMeta ItemMeta matches snapshot for pending 1`] = `
<table
  className="wrapper"
>
  <tbody>
    <tr>
      <td>
        Added
      </td>
      <td>
        3/29/2020, 2:30:45 PM
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`components/ItemMeta ItemMeta matches snapshot for running 1`] = `
<table
  className="wrapper"
>
  <tbody>
    <tr>
      <td>
        Added
      </td>
      <td>
        3/29/2020, 2:30:45 PM
      </td>
    </tr>
    <tr>
      <td>
        Updated
      </td>
      <td>
        3/29/2020, 2:32:30 PM
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`components/ItemMeta ItemMeta matches snapshot for succeeded 1`] = `
<table
  className="wrapper"
>
  <tbody>
    <tr>
      <td>
        Added
      </td>
      <td>
        3/29/2020, 2:30:45 PM
      </td>
    </tr>
    <tr>
      <td>
        Completed
      </td>
      <td>
        3/29/2020, 2:32:30 PM
      </td>
    </tr>
  </tbody>
</table>
`;
